src/marlin_stubs/sdcard/M20-M30_M32-M34.cpp

```diff
@@ -30,7 +30,15 @@ void GcodeSuite::M20() {
  * M21 - Initialize SD card
  */
 void GcodeSuite::M21() {
-    // not necessary - empty implementation
+    // required for Octoprint / third party tools to simulate an inserted SD card when using USB
+    DIR *dir;
+    dir = opendir("/usb/");
```

Comment:
There's an easier way to check if the USB is present - ˙marlin_vars().media_inserted˙

Reply:
Perfect, exactly what I was searching for. Changed the PR accordingly, thx !